cfpb/consumerfinance.gov
# Deprecate prepaid search custom breadcrumbs

The prepaid agreements search app currently includes some custom breadcrumb logic to try to improve the user experience when going from a search result to a detail page and back. This logic breaks in unexpected ways when pages are cached in Akamai. This commit removes this custom logic.

The current and originally desired behavior is this:

1. A user visits the prepaid agreements search page at https://www.consumerfinance.gov/data-research/prepaid-accounts/search-agreements/.
2. Without applying any filters, the user clicks on a result detail, for example https://www.consumerfinance.gov/data-research/prepaid-accounts/search-agreements/detail/72669/.
The "Prepaid Product Agreements Database" breadcrumb links back to the search page, https://www.consumerfinance.gov/data-research/prepaid-accounts/search-agreements/.
3. The user applies a filter, for example https://www.consumerfinance.gov/data-research/prepaid-accounts/search-agreements/?prepaid_type=Student.
3. The user clicks on a result detail, for example, https://www.consumerfinance.gov/data-research/prepaid-accounts/search-agreements/detail/48466/.
The custom breadcrumb logic uses the HTTP referrer header to customize the "Prepaid Product Agreements Database" breadcrumb so that it links back to a _filtered_ version of the search page, https://www.consumerfinance.gov/data-research/prepaid-accounts/search-agreements/?prepaid_type=Student.

The problem with this logic has to do with the Akamai caching that fronts www.consumerfinance.gov. If detail pages don't exist in the Akamai cache (which is typical, as detail pages may not be accessed often), then we risk having them get baked into the cache with customized breadcrumb links.

So, for example, the above set of actions by some User 1 would result in Akamai caching a version of the detail URL https://www.consumerfinance.gov/data-research/prepaid-accounts/search-agreements/detail/48466/ with a breadcrumb pointing to a filtered search.

If a second User 2 now visits that same detail URL either without filtering or via some other combination of filters, the breadcrumb link is now wrong.

Because users already have a Back button in their browsers, this customization of the breadcrumb is not the only way for them to get back to their original search.

An additional note about this custom logic: we don't attempt to do this kind of breadcrumb modification from other search results pages. For example, if a user visits the blog page (https://www.consumerfinance.gov/about-us/blog/), filters, and then
visits a specific blog post, the breadcrumb link "Blog" goes back to the regular unfiltered blog index. So removal of this custom prepaid breadcrumb logic helps standardize user experience across the site.

Ping @wpears for your oversight / input. We've discussed this proposed change on EC.

## How to test this PR

With a local server, visit http://localhost:8000/data-research/prepaid-accounts/search-agreements/ and experiment with filters and see how the detail breadcrumb link stays constant.

## Checklist
